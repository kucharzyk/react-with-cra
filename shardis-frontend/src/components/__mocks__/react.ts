import * as React from 'react';

// Temporary hack to suppress error
// https://github.com/facebookincubator/create-react-app/issues/3199
window.requestAnimationFrame = function (callback) {
  setTimeout(callback, 0);
  return 0;
};

module.exports = React;
