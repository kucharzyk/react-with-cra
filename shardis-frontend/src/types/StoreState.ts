import { CounterState } from '../reducers/counter';

export default interface StoreState {
  counter: CounterState;
}
