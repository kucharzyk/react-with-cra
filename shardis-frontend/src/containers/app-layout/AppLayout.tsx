import * as React from 'react';
import { StatelessComponent } from 'react';
import withStyles from 'material-ui/styles/withStyles';
import { WithStyles } from 'material-ui';
import { StyleRulesCallback, Theme } from 'material-ui/styles';
import AppHeader from '../../components/app-header/AppHeader';
import AppFooter from '../../components/app-footer/AppFooter';
import AppRoutes from '../app-routes/AppRoutes';

const styles: StyleRulesCallback<'root'> = (theme: Theme) => ({
  root: {
    padding: theme.spacing.unit
  },
});

const AppLayout: StatelessComponent<{} & WithStyles<'root'>> = (props) => (
  <div className={props.classes.root}>
    <AppHeader/>
    <AppRoutes/>
    <AppFooter/>
  </div>
);

export default withStyles(styles)<{}>(AppLayout);
