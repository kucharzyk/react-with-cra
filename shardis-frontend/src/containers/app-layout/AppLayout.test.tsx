import * as React from 'react';
import { StatelessComponent } from 'react';

import AppLayout from './AppLayout';
import { StaticRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import StoreState from '../../types/StoreState';
import { Provider } from 'react-redux';
import createShallow from 'material-ui/test-utils/createShallow';
import toJson from 'enzyme-to-json';

describe('AppLayout', () => {

  let WrappedComponent: StatelessComponent;

  beforeEach(() => {

    const mockStore = configureStore<StoreState>();
    const store = mockStore({
      counter: 1
    });

    WrappedComponent = () => (
      <Provider store={store}>
        <StaticRouter location={'/'} context={{}}>
          <AppLayout/>
        </StaticRouter>
      </Provider>
    );
  });

  it('renders snapshot', () => {
    const shallowComponent = createShallow({untilSelector: 'AppLayout'})(<WrappedComponent/>);
    expect(toJson(shallowComponent)).toMatchSnapshot();
  });

});
