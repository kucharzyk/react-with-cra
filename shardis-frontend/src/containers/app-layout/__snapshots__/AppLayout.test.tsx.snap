// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout renders snapshot 1`] = `
<div
  className="AppLayout-root-1"
>
  <withRouter(WithStyles(AppHeader)) />
  <WithStyles(AppRoutes) />
  <WithStyles(AppFooter) />
</div>
`;
